src/main/java/com/epam/reportportal/extension/azure/AzureExtension.java

```diff
@@ -184,4 +224,165 @@ private void removeListeners() {
+		String organizationUrl = params.get("url").toString();
+		String projectName = params.get("project").toString();
+		String personalAccessToken = params.get("oauthAccessKey").toString();
+
+		ApiClient defaultClient = getConfiguredApiClient(personalAccessToken);
+		String organizationName = organizationUrl.replace(defaultClient.getBasePath(), "");
+
+		ClassificationNodesApi nodesApi = new ClassificationNodesApi(defaultClient);
+		List<WorkItemClassificationNode> areaNodes = new ArrayList<>();
+		List<WorkItemClassificationNode> iterationNodes = new ArrayList<>();
+		try {
+            List<WorkItemClassificationNode> nodes = nodesApi
+                    .classificationNodesGetRootNodes(organizationName, projectName, API_VERSION, DEPTH);
+			for (WorkItemClassificationNode node : nodes) {
+				if (node.getStructureType().equals("area")) {
+					areaNodes = extractNestedNodes(node);
+				} else if (node.getStructureType().equals("iteration")) {
+					iterationNodes = extractNestedNodes(node);
+				}
+			}
+		} catch (ApiException e) {
+            LOGGER.error("Unable to load classification nodes: " + e.getMessage(), e);
+			throw new ReportPortalException(ErrorType.UNABLE_INTERACT_WITH_INTEGRATION,
+					"Unable to load classification nodes. Code: " + e.getCode() + ", Message: " + e.getMessage(), e);
+		}
+
+		WorkItemTypesFieldApi issueTypeFieldsApi = new WorkItemTypesFieldApi(defaultClient);
+		FieldsApi fieldsApi = new FieldsApi(defaultClient);
+        List<PostFormField> ticketFields = new ArrayList<>();
+		try {
+            List<WorkItemTypeFieldWithReferences> issueTypeFields = issueTypeFieldsApi
+                    .workItemTypesFieldList(organizationName, projectName, issueType, API_VERSION, EXPAND);
+
+            for (WorkItemTypeFieldWithReferences field : issueTypeFields) {
+				WorkItemField detailedField = null;
+                // Skip fields that return 404 on request
+				try {
+					detailedField = fieldsApi
+							.fieldsGet(organizationName, field.getReferenceName(), projectName, API_VERSION);
+				} catch (ApiException e) {
+					continue;
+				}
+
+                // Skip read-only fields
+				if (detailedField.isReadOnly()) {
+					continue;
+				}
+
+                List<AllowedValue> allowed = new ArrayList<>();
+				switch (detailedField.getName()) {
+                    // Skip Work Item Type field cause we have the same custom field
+					case "Work Item Type":
+						continue;
+					case "Iteration ID":
+						for (WorkItemClassificationNode node : iterationNodes) {
+							allowed.add(new AllowedValue(node.getId().toString(), node.getName()));
+						}
+						break;
+					case "Area ID":
+						for (WorkItemClassificationNode node : areaNodes) {
+							allowed.add(new AllowedValue(node.getId().toString(), node.getName()));
+						}
+						break;
+					default:
+						for (Object value : field.getAllowedValues()) {
+							allowed.add(new AllowedValue(value.toString().replace(" ", "_"), value.toString()));
```

Comment:
DRY. This is small but meaningful .replace(" ", "_"). better to make a function with good name